linuxcomport.cpp

```diff
@@ -7,91 +7,98 @@
 
 #include "linuxcomport.h"
 
-void WaitForAnswer(int fileDescriptor) {
-	fd_set rfds;
-	FD_ZERO(&rfds);
-	FD_SET(0, &rfds);
-	struct timeval timeout;
-	timeout.tv_sec = 0;
-	timeout.tv_usec = 50000;
-	select(fileDescriptor, &rfds, NULL, NULL, &timeout);
+void WaitForAnswer(int fileDescriptor)
+{
+    fd_set rfds;
+    FD_ZERO(&rfds);
+    FD_SET(0, &rfds);
+    struct timeval timeout;
+    timeout.tv_sec = 0;
+    timeout.tv_usec = 50000;
+    select(fileDescriptor, &rfds, NULL, NULL, &timeout);
 }
 
-size_t linuxComPort::LinuxComPort::sendData(const char* buffer, const size_t length) {
-	size_t count = write(fileDescriptor, buffer, length);
-	return count == length;
+size_t linuxComPort::LinuxComPort::sendData(const char* buffer, const size_t length)
+{
+    size_t count = write(fileDescriptor, buffer, length);
+    return count == length;
 }
 
-size_t linuxComPort::LinuxComPort::sendByte(char data, bool sendInverted) {
-	uint count = 0;
+size_t linuxComPort::LinuxComPort::sendByte(char data, bool sendInverted)
+{
+    uint count = 0;
 
-	char c[] = { (char) data, (char) ~data };
+    char c[] = { (char)data, (char)~data };
 
-	if (sendInverted) {
-		count += write(fileDescriptor, c, 2);
-	} else {
-		count += write(fileDescriptor, c, 1);
-	}
+    if (sendInverted) {
+        count += write(fileDescriptor, c, 2);
+    } else {
+        count += write(fileDescriptor, c, 1);
+    }
 
-	return count > 0;
+    return count > 0;
 }
 
-size_t linuxComPort::LinuxComPort::close() {
-	fileDescriptor = -1;
-	return true;
+size_t linuxComPort::LinuxComPort::close()
+{
+    fileDescriptor = -1;
+    return true;
 }
 
-size_t linuxComPort::LinuxComPort::getComPortStatus() {
-	return fileDescriptor > 0;
+size_t linuxComPort::LinuxComPort::getComPortStatus()
+{
+    return fileDescriptor > 0;
 }
 
-size_t linuxComPort::LinuxComPort::receiveData(char* buffer, size_t* length) {
-	//WaitForAnswer(fileDescriptor);
-	ssize_t res = read(fileDescriptor, buffer, (unsigned long)*length);
+size_t linuxComPort::LinuxComPort::receiveData(char* buffer, size_t* length)
+{
+    //WaitForAnswer(fileDescriptor);
+    ssize_t res = read(fileDescriptor, buffer, (unsigned long)*length);
```

Comment:
hoppla, was machst du den hier? du bekommst nach dieser funktion ein signed size_t und wandelst es ohne pruefung in ein size_t um. Somit wird die laenge komplett falsch zugewiesen. Wenn der anwender den Returnwert prueft, erhaelt er zwar ein false, aber mal Hand aufs Herz, wie oft prueft man schon die return werte jeder funktion?